libs/blocks/merch/merch.js

```diff
@@ -37,16 +37,33 @@ export async function loadEntitlements() {
     fetch(entitlementsURL),
   ]).then(([{ default: getUserEntitlements }, mappings]) => {
     if (!mappings.ok) return [];
-    return Promise.all([getUserEntitlements(), mappings.json()]);
+    return Promise.all([getUserEntitlements({ params: [{ name: 'include', value: 'OFFER.PRODUCT_ARRANGEMENT' }], format: 'raw' }), mappings.json()]);
   });
   return loadEntitlements.promise;
 }
 
 async function getCheckoutAction(offers) {
-  const [{ arrangement_codes: aCodes } = {}, entitlementsMappings] = await loadEntitlements();
-  if (aCodes === undefined) return undefined;
+  let upgradeAction;
+  const [entitlements, entitlementsMappings] = await loadEntitlements();
+  const aCodes = entitlements?.map((offer) => offer.offer.product_arrangement_code);
```

Comment:
is `product_arrangement_code` always supposed to be here?

Reply:
no, this is download flow (different feature), will be fixed in a separate ticket
@yesil should have a jira number